### PR TITLE
[SMALLFIX] Add description in method and add some UT in storeMeta

### PR DESCRIPTION
**Pull request by yabola**
1. add some description on methond (`getBlockList()` and `getBlockListByStorageLocation` ) in `BlockStoreMeta` to make it more clearer.
2. add some UT for `BlockStoreMeta`

fix
 https://github.com/Alluxio/new-contributor-tasks/issues/588
 https://github.com/Alluxio/new-contributor-tasks/issues/589

### CODE REVIEW

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 1]
```diff
@@ -90,6 +90,7 @@
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_QUOTA, "2GB")
           .put(PropertyKey.WORKER_TIERED_STORE_LEVEL1_DIRS_PATH, mHddDir)
           .put(PropertyKey.WORKER_RPC_PORT, "0")
+          .put(PropertyKey.WORKER_MANAGEMENT_TIER_ALIGN_RESERVED_BYTES, "512MB")
```
Comment (yabola) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:93]: The default value of 1024M is not obvious enough , which will affect the allocation of blocks

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 2]
```diff
@@ -212,11 +213,28 @@ public void getReport() {
...
+    assertEquals(2, storeMeta.getBlockListByStorageLocation().size());
+    assertEquals(0, storeMeta.getBlockList().get("MEM").size());
+    // 1GB (`MEM`) + 2GB (`HDD`)
+    assertEquals(3221225472L, storeMeta.getCapacityBytes());
+    // 512MB * 2 (reserved space) + 1bit (`MEM`) + 1GB (`HDD`)
+    assertEquals(2147483649L, storeMeta.getUsedBytes());
+    mBlockWorker.commitBlock(sessionId, blockId1, true);
+    mBlockWorker.commitBlock(sessionId, blockId2, true);
+    storeMeta = mBlockWorker.getStoreMetaFull();
+    assertEquals(1, storeMeta.getBlockList().get("MEM").size());
```
Comment (yabola) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:236]: we can't assert usedBytes after commit block cause the real path size is 0 in UT.
see `BlockMetadataManager#commitTempBlockMeta()` -> `dir.addBlockMeta(block) // block size is 0 here`

Comment (yabola) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:236]: we can't assert usedBytes after commit block cause the real path size is 0 in UT.
see BlockMetadataManager#commitTempBlockMeta() -> dir.addBlockMeta(block) // block size is 0 here

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 1 above)
Comment (yabola) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:93]: The default value of 1024M is not obvious enough , which will affect the allocation of blocks

**Comment by yabola** [comment]
@LuQQiu please take a look~

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java [hunk 3]
```diff
@@ -61,13 +61,23 @@
 
   @Override
   public Map<String, List<Long>> getBlockList() {
```
Comment (LuQQiu) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java:63]: @yabola it's always to put the comment in the getBlockList() method declaration. That's where people will usually view to understand the method. And the comment should be implementation independent and as a contract between caller and implementor.
For example, you can change the BlockStoreMeta.getBlockList description:
```
This method is only available when users initialize the blockStoreMeta with full block meta requirement
```

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java [hunk 4]
```diff
@@ -61,13 +61,23 @@
...
+     * will not be initialized and be `NULL` value.
+     */
     Preconditions.checkNotNull(mBlockIdsOnTiers, "mBlockIdsOnTiers");
 
     return mBlockIdsOnTiers;
   }
 
   @Override
   public Map<BlockStoreLocation, List<Long>> getBlockListByStorageLocation() {
+    /**
```
Comment (LuQQiu) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java:76]: Same for this method description

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 5]
```diff
@@ -180,9 +181,9 @@ public void commitBlockOnRetry() throws Exception {
   public void createBlock() throws Exception {
     long blockId = mRandom.nextLong();
     long sessionId = mRandom.nextLong();
-    long initialBytes = 1;
+    long initialBytes = 1024 * 1024 * 1024;
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:184]: Can you use Constants.MB? Can we not initialize a large block store  e.g. https://github.com/Alluxio/alluxio/pull/13375?

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 6]
```diff
@@ -212,11 +213,28 @@ public void getReport() {
...
+  public void getStoreMeta() throws Exception {
     mBlockWorker.getStoreMeta();
-    mBlockWorker.getStoreMetaFull();
     verify(mBlockStore, times(2)).getBlockStoreMeta(); // 1 is called at metrics registration
-    verify(mBlockStore).getBlockStoreMetaFull();
+    long blockId1 = mRandom.nextLong();
+    long blockId2 = mRandom.nextLong();
+    long sessionId = mRandom.nextLong();
+    mBlockWorker.createBlock(sessionId, blockId1, 0, "", 1L);
+    // Available space in `MEM` is not enough. Block will be allocated in `HDD`.
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:223]: That's not true. It's related to the `put(PropertyKey.WORKER_MANAGEMENT_TIER_ALIGN_RESERVED_BYTES, "0")` which  if reserved bytes is bigger than mem tier size, the blocks will not be allocated in the mem tier

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 7]
```diff
@@ -212,11 +213,28 @@ public void getReport() {
...
+    mBlockWorker.createBlock(sessionId, blockId2, 0, "", 1024 * 1024 * 1024L);
+    BlockStoreMeta storeMeta = mBlockWorker.getStoreMetaFull();
+    assertEquals(2, storeMeta.getBlockList().size());
+    assertEquals(2, storeMeta.getBlockListByStorageLocation().size());
+    assertEquals(0, storeMeta.getBlockList().get("MEM").size());
+    // 1GB (`MEM`) + 2GB (`HDD`)
+    assertEquals(3221225472L, storeMeta.getCapacityBytes());
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:230]: Please use some easy to understanding values, e.g.Constants.GB * 3

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 8]
```diff
@@ -212,11 +213,28 @@ public void getReport() {
...
+    // 512MB * 2 (reserved space) + 1bit (`MEM`) + 1GB (`HDD`)
+    assertEquals(2147483649L, storeMeta.getUsedBytes());
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:232]: Same here and remove the comment

**Review by LuQQiu — COMMENTED** [review]
Thanks for improving the comments and tests, left some comments

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java (hunk 3 above)
Comment (yabola) [reply core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockStoreMeta.java]: done

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 5 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: done

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 7 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: done

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 6 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: I check in `DefaultStorageDir` -> `mAvailableBytes = capacityBytes - reservedBytes` ,
available space means that the reserved space is removed. allocator will judge whether there is enough space based on AvailableBytes. so i think this description is right...

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 9]
```diff
@@ -211,11 +217,26 @@ public void getReport() {
   }
 
   @Test
-  public void getStoreMeta() {
+  public void getStoreMeta() throws Exception {
     mBlockWorker.getStoreMeta();
-    mBlockWorker.getStoreMetaFull();
     verify(mBlockStore, times(2)).getBlockStoreMeta(); // 1 is called at metrics registration
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:222]: No need to do in this PR, but will be better if you can do the similar changes against getStoreMeta just like what you did for `getStoreMetaFull`

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 10]
```diff
@@ -176,9 +176,15 @@ public void commitBlockOnRetry() throws Exception {
...
     long blockId = mRandom.nextLong();
     long sessionId = mRandom.nextLong();
-    long initialBytes = 1;
+    long initialBytes = 1L;
     String path = mBlockWorker.createBlock(sessionId, blockId, 0, "", initialBytes);
-    assertTrue(path.startsWith(mMemDir)); // tier 0 is mem
+    assertTrue(path.startsWith(mMemDir));
+
+    long blockId2 = mRandom.nextLong();
+    long initialBytes2 = 1L * Constants.GB;
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:184]: In this test, try not to set a large initial bytes number. This test will actually create a block store. With the large initial bytes, the worker will actually try to allocate 1GB or 2GB in the system. When this test is running in a env with little disk space available, this test may be failed.

All the unit tests are better to be designed as light-weighted tests

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java [hunk 11]
```diff
@@ -211,11 +217,26 @@ public void getReport() {
...
     mBlockWorker.getStoreMeta();
-    mBlockWorker.getStoreMetaFull();
     verify(mBlockStore, times(2)).getBlockStoreMeta(); // 1 is called at metrics registration
-    verify(mBlockStore).getBlockStoreMetaFull();
+    long blockId1 = mRandom.nextLong();
+    long blockId2 = mRandom.nextLong();
+    long sessionId = mRandom.nextLong();
+    mBlockWorker.createBlock(sessionId, blockId1, 0, "", 1L);
+    // Available space in MEM is not enough. Block will be allocated in HDD.
+    mBlockWorker.createBlock(sessionId, blockId2, 0, "", 2L * Constants.GB);
```
Comment (LuQQiu) [inline core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java:228]: Same, don't set initial size as a large number

**Review by LuQQiu — CHANGES_REQUESTED** [review]
Thanks for improving the description and docs! 

Can you modify the PR description to include the main points of what you did in this PR?
The PR description will be included in the log commit message for other people to know what's the content for the commit.

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 10 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: done

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 9 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: done

File: core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java (hunk 11 above)
Comment (yabola) [reply core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java]: done

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for improving the description and the unit tests!

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**